lib/improver/blending/weights.py

```diff
@@ -238,12 +242,565 @@ def process_coord(cube, coordinate, coord_exp_vals=None,
+
+                **target_points** (np.ndarray):
+                    Points within the cube that will be the target points
+                    for the interpolation.
+
+                **source_weights** (np.ndarray):
+                    Weights from the configuration dictionary that will be
+                    used as the input to the interpolation.
+
+                **axis** (int):
+                    Axis of self.weighting_coord_name within the input cube.
+                    This will be used to define the axis of interpolation.
+
+                **fill_value** (tuple):
+                    Values that be used if extrapolation is required. The
+                    fill values will be used as constants that are extrapolated
+                    if the target_points are outside the source_points
+                    provided. These are equal to the weight associated with the
+                    first and last values along the weighting coord e.g.
+                    forecast_period.
+
+        """
+        source_points = weights_cube.coord(self.weighting_coord_name).points
+        target_points = cube.coord(self.weighting_coord_name).points
+        source_weights = weights_cube.core_data()
+        axis, = weights_cube.coord_dims(self.weighting_coord_name)
+
+        coord_values = (
+            {self.weighting_coord_name: lambda cell: cell == source_points[0]})
+        constr = iris.Constraint(coord_values=coord_values)
+        lower_fill_value = weights_cube.extract(constr).core_data()
+
+        coord_values = (
+            {self.weighting_coord_name:
+             lambda cell: cell == source_points[-1]})
+        constr = iris.Constraint(coord_values=coord_values)
+        upper_fill_value = weights_cube.extract(constr).core_data()
+
+        fill_value = (lower_fill_value, upper_fill_value)
+        return source_points, target_points, source_weights, axis, fill_value
+
+    @staticmethod
+    def _create_coord_and_dims_list(
+            base_cube, cube_with_exception_coord, coord_list,
+            exception_coord_name):
+        """Create a list of coordinates and their dimensions, primarily
+        for usage when constructing an iris.cube.Cube.
+
+        Args:
+            base_cube (iris.cube.Cube):
+                Cube from which all the coordinates within the coord_list
+                will be copied, apart from the coordinate defined by the
+                exception_coord_name, which will be taken from the
+                cube_with_exception_coord.
+            cube_with_exception_coord (iris.cube.Cube):
+                Cube from which the exception coordinate will be taken.
+            coord_list (iterable):
+                List/tuple of coordinates.
+            exception_coord_name (str):
+                Name of the exception coordinate that will be taken from the
+                cube_with_exception_coord.
+
+        Returns:
+            new_coord_list (list):
```

Comment:
This function actually returns a list of tuples where each tuple is (coord, index_of_coord).

Reply:
I've updated this docstring.